lib/pages/detection/annotation.dart

```diff
@@ -161,26 +96,50 @@ class _AnnotationPageState extends State<AnnotationPage> {
                           width: 300,
                           height: 300,
                           child: FreeDraw(
-                            key: _freeDrawKey,
                             imageLink: snapshot.data as String,
                           ),
                         ),
                       );
                     }
                   }),
+              Text('Current Label: ${notifier.label}'),
               ElevatedButton(
-                  onPressed: () {
-                    _showPopup();
-                  },
-                  child: Text("Label Annotation",
-                      style: textTheme.labelLarge!.copyWith(
-                        color: Theme.of(context).colorScheme.onPrimary,
-                      ))),
+                onPressed: () {
+                  GoRouter.of(context)
+                      .push("/main/detection/${widget.detectionId}/label");
+                },
+                child: Text(
+                  "Select Label",
+                  style: textTheme.labelLarge!.copyWith(
+                    color: Theme.of(context).colorScheme.onPrimary,
+                  ),
+                ),
+              ),
+              ElevatedButton(
+                onPressed: () {
+                  if (notifier.isCompleteAnnotation()) {
+                    notifier.addToAllAnnotations();
+                  } else {
+                    String message;
+                    if (notifier.label == null) {
+                      message = "Please Enter A Label For Current Annotation";
```

Comment:
The capitalization should be improved/decided
```suggestion
                      message = "Please Enter a Label for Current Annotation";
```